lib/simp/rake/build/constants.rb

```diff
@@ -19,6 +19,12 @@ def init_member_vars( base_dir )
     @build_version = version || Facter.fact('operatingsystemmajrelease').value
     @build_arch = arch || Facter.fact('architecture').value
 
+    # Working around the SIMP::RPM.system_dist workaround
+    if @build_distro =~ /CentOS|RedHat/i
+      @build_rpm_dist = "el#{@build_version}"
+      ENV['SIMP_RPM_dist'] = @build_rpm_dist
```

Comment:
Don't you only want to set the environment variable if it is not already set? Also, why isn't this info built into the already existing SIMP_BUILD_distro environment variable?

Reply:
@op-ct Looks like @lnemsick-simp has a good point. If you pass `SIMP_BUILD_distro=CentOS,7` then all of the information required should be passed along

Reply:
@lnemsick-simp No.  This `SIMP_RPM_dist` environment variable assignment is necessary because `SIMP::RPM` _must_ agree with whatever the rest of the SIMP build is, otherwise the SIMP build process will produce packages with inconsistent `dist` macros, and will eventually fail to repoclose.

A deeper reason is that, `SIMP_BUILD_distro` variable no longer determines the `dist` macro for all the packages `simp-core` builds.  Most components will build using `@build_version`, but some top-level of the assets will follow `@simp_version`, which is assigned on line 35 via a `SIMP::RPM.new` on `simp.spec`, which—since the docker `system_dist` hack—always return the host OS.

Another reason is that ` SIMP::RPM` doesn't (and shouldn't) know anything about `Simp::Rake::Build::Constants`, which is why it exposes the `SIMP_RPM_dist` in the first place.  `Simp::Rake::Build::Constants` uses it to keep the build interpretation of .spec files consistent—and so can any other code that uses `SIMP::RPM`.

@trevor-vaughan  _If_ the environment variable `SIMP_BUILD_distro` is set, the distro version _may_ be specified—but it doesn't have to be.

`SIMP_BUILD_distro` is a comma-delimited string with a maximum of 3 values for distro (name), distro (major version), and architecture.  But at a minimum, `SIMP_BUILD_distro` might only specify the distro name (e.g, `CentOS`) and omit the distro version. ‡

`SIMP_RPM_dist` also uses the RPM `dist` macro format (`el7`) instead bare number (`7`) directly from `SIMP_BUILD_dist`.

This is why `SIMP_RPM_dist` is assigned to `@build_rpm_dist`, which will always have an appropriate value in the appropriate format.



----
‡ = In case you're wondering, this minimal specification of  `SIMP_BUILD_distro` was by design: The original `rake simp:auto` could scan a directory of ISOs and use the release mappings data to build every matching version of SIMP it could find, unless restricted by `SIMP_BUILD_distro`.  Subsequent ninja hacks ended up breaking that capability (and yet more hacks attempted to "fix" with the {redundant and } `build_metadata.yaml` file)—but we'll get to that in due course.